### PR TITLE
Add parameter to `CCDData.to_hdu` to get an `ImageHDU` from a `CCDData` object

### LINKED ISSUE #12933: Convert CCDData to ImageHDU
### Description
As far as I can tell, currently there's no way to directly convert a `CCDData` object to an `ImageHDU` object. If we write it to a file using `CCDData.write()` it will always create a file where the first HDU is a `PrimaryHDU` that contains the `CCDData.data`, followed by optionally some `ImageHDU`s that contain mask or uncertainty. If we instead use `CCDData.to_hdu()`, it will return an `HDUList` equivalent to the file it writes with `CCDData.write()`, that is, the `CCDData.data` is stored in the first element of the `HDUList`, which is always a `PrimaryHDU`.

This is somewhat unexpected given that you can already do it the other way around (that is, convert a `ImageHDU` object to a `CCDData` object):

```python
fits.HDUList([
    fits.PrimaryHDU(),
    fits.ImageHDU(data=np.ones((2, 2))),
    fits.ImageHDU(data=np.ones((5, 5)), header=fits.Header({'BUNIT': 'm'})),
]).writeto('test.fits')  # create example file

ccd_image = CCDData.read('test.fits', hdu=2)  # you can successfully read the 5x5 ImageHDU
```
The problem is that if we then want to use this `ccd_image` as an extension to another FITS file, there's no obvious way to get an `ImageHDU` which would allow us to do that.  As far as I can tell, there's also no trivial way to convert a `PrimaryHDU` to a `ImageHDU`. We could manually create a new `ImageHDU` by copying the data from the `PrimaryHDU`, as well as its relevant cards and so on... but this seems unnecessarily complicated.

I propose the following interfaces:

```python
# Option A: add a new parameter to CCDData.to_hdu() for this functionality
hdus = ccd_image.to_hdu(as_image_hdu=True)  # This returns a HDUList where the first element is an ImageHDU instead of a PrimaryHDU

# Option B: create a new convenience function
hdu = fits.ccddata_to_image_hdu(ccd_image) # This returns a single ImageHDU

# Option C: allowing the user to append the image to an existing FITS file
ccd_image.write('test.fits', append=True) # appends original ImageHDU to existing file
```

### Additional context
This seems somewhat similar to the situation with `Table` and `BinTableHDU`. In that case, we can also specify an `hdu` parameter when reading:

```python
fits.BinTableHDU.from_columns([
    fits.Column(name='test', format='J', array=(1, 2))
]).writeto('table.fits')  # creates a new file with a PrimaryHDU followed by this BinTableHDU
t = Table.read('table.fits', hdu=1) # reads the BinTableHDU as a Table
```

From here we can use:

```python
t.write('new_table.fits')  #  creates a new file with a PrimaryHDU followed by the original BinTableHDU
t.write('existing_table.fits', append=True)  # appends original BinTableHDU to existing file
hdu = fits.table_to_hdu(t)  # returns original BinTableHDU
```

### PR DESCRIPTION
**Pull request by kYwzor**
### Description
This pull request is to address the lack of a trivial way to get an `ImageHDU` from a `CCDData` object. It does so by adding a new `as_image_hdu` parameter to `CCDData.to_hdu`. The parameter is also added to `fits_ccddata_writer`, which passes it on to `CCDData.to_hdu` and inserts an empty `PrimaryHDU`.

Fixes #12933

Note: This pull request likely needs some work, especially in the  docstrings and documentation.

### Checklist for package maintainer(s)
This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Please add a change log under "feature" category; see https://github.com/astropy/astropy/blob/main/docs/changes/README.rst

I'll let subpackage maintainers decide if this also needs update to user facing doc.

File: docs/changes/nddata/12962.feature.rst
Comment (saimn) [inline docs/changes/nddata/12962.feature.rst:1]: and for ``CCDData.write``.

File: docs/changes/nddata/12962.feature.rst
Comment (saimn) [inline docs/changes/nddata/12962.feature.rst:4]: The last sentence (*Before this feature ...*) seems unnecessary, it 's a new option so it means it wasn't there before.

**Review by saimn — APPROVED** [review]
Thanks @kYwzor, seems good to me (just 2 little comments).

**Comment by kYwzor** [comment]
@saimn thank you for the feedback, I believe I've addressed your comments with the latest commit.

**Comment by mwcraig** [comment]
Thank you again for this, @kYwzor, and for the review @saimn. Merging since the failures are unrelated.

**Pull request merged**
